client.go

```diff
@@ -0,0 +1,331 @@
+package iprepd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Client is the iprepd service client
+type Client struct {
+	hostURL    string
+	authStr    string
+	httpClient *http.Client
+}
+
+// in an effort to keep the client's error handling consistent and for ease of
+// testing, all error messages should be listed below and reused where possible
+const (
+	// input validation errors
+	clientErrURLEmpty            = "url cannot be empty"
```

Comment:
NB: Might be nice to expose all of these client errors for use by users of the clients. Could be helpful with targeted error handling.

Reply:
I agree -- lets leave that for a follow up PR where we discuss error handling in the client as a whole, including how much of the server side errors we want to expose to the user